enterprise_access/apps/api/v1/views/browse_and_request.py

```diff
@@ -1022,6 +1036,135 @@ def approve(self, request, *args, **kwargs):
+                    results["not_found"].append(
+                        {"uuid": uuid_val, "detail": "Request not found"}
+                    )
+
+        for uuid_val, lc_request in requests_by_uuid.items():
+            # For approve_all mode, we already filtered to REQUESTED state
+            # For specific UUID mode, check state and skip if not REQUESTED
+            if (not approve_all and lc_request.state != SubsidyRequestStates.REQUESTED):
+                results["skipped"].append(
+                    {"uuid": uuid_val, "state": lc_request.state}
+                )
+                continue
+
+            learner_email = lc_request.user.email
+            content_key = lc_request.course_id
+            content_price_cents = lc_request.course_price
+
+            lc_request_action = LearnerCreditRequestActions.create_action(
+                learner_credit_request=lc_request,
+                recent_action=get_action_choice(
+                    SubsidyRequestStates.APPROVED
+                ),
+                status=get_user_message_choice(SubsidyRequestStates.APPROVED),
+            )
+
+            try:
+                with transaction.atomic():
+                    assignment = approve_learner_credit_request_via_policy(
+                        policy_uuid,
+                        content_key,
+                        content_price_cents,
+                        learner_email,
+                        lc_request.user.lms_user_id,
+                    )
+                    lc_request.assignment = assignment
+                    lc_request.save()
+                    lc_request.approve(request.user)
+                    send_learner_credit_bnr_request_approve_task.delay(
+                        assignment.uuid
+                    )
+                results["approved"].append(
+                    serializers.LearnerCreditRequestSerializer(
+                        lc_request
+                    ).data
+                )
+            except SubisidyAccessPolicyRequestApprovalError as exc:
+                error_msg = (
+                    f"[LC REQUEST BULK APPROVAL] Failed to approve learner credit request "
+                    f"with UUID {uuid_val}. Reason: {exc.message}."
+                )
+                logger.exception(error_msg)
+                # Update action with error
+                lc_request_action.status = get_user_message_choice(
+                    SubsidyRequestStates.REQUESTED
+                )
+                lc_request_action.error_reason = get_error_reason_choice(
+                    LearnerCreditRequestActionErrorReasons.FAILED_APPROVAL
+                )
+                lc_request_action.traceback = format_traceback(exc)
+                lc_request_action.save()
+                results["failed"].append(
+                    {"uuid": uuid_val, "detail": exc.message}
+                )
+
+        return Response(results, status=status.HTTP_200_OK)
```

Comment:
Can we limit the data that we are returning to frontend? I believe just the UUID, state, and detail (error) message should be enough.